crates/dbsp/src/circuit/dbsp_handle.rs

```diff
@@ -442,25 +457,48 @@ impl DBSPHandle {
         // Send command.
         for (worker, sender) in self.command_senders.iter().enumerate() {
             if sender.send(command.clone()).is_err() {
-                let _ = self.kill_inner();
-                return Err(DBSPError::Runtime(RuntimeError::WorkerPanic { worker }));
+                let panic_info = self.collect_panic_info().unwrap_or_default();
+
+                // Worker thread panicked. Exit without waiting for all workers to exit
+                // to avoid deadlocks due to workers waiting for each other.
+                self.kill_async();
+                return Err(DBSPError::Runtime(RuntimeError::WorkerPanic { panic_info }));
             }
             self.runtime.as_ref().unwrap().unpark_worker(worker);
         }
 
+        // Use `Select` to wait for responses from all workers simultaneously.
+        // This way if one of the workers panics, leaving other workers waiting
+        // for it in exchange operators, we won't deadlock waiting for these
+        // workers.
+        let mut select = Select::new();
+        for receiver in self.status_receivers.iter() {
+            select.recv(receiver);
+        }
+
+        let mut expected_responses = self.status_receivers.len();
+
         // Receive responses.
-        for (worker, receiver) in self.status_receivers.iter().enumerate() {
-            match receiver.recv() {
+        while expected_responses > 0 {
```

Comment:
This is petty, but I'd probably write this as a `for`:
```
for _ in 0..self.status_receivers.len() {
...
}
```
rather than a `loop`.

Reply:
Good point!